src/lib.rs

```diff
@@ -881,12 +874,24 @@ impl Build {
     /// Panics if `output` is not formatted correctly or if one of the underlying
     /// compiler commands fails. It can also panic if it fails reading file names
     /// or creating directories.
-    pub fn compile(&self, output: &str) {
+    pub fn compile(&mut self, output: &str) {
```

Comment:
Unfortunately taking `&mut self` here is a breaking change (even if I suspect it won’t affect many if any users in practice). Could `resolve_supported_flags` return a new Vec instead of mutating the existing one?

Reply:
This suggestion ended up requiring even greater interface changes, so I've abandoned it.